### PR TITLE
Report statistics to the automated leaderboard

### PR DESCRIPTION
**Pull request by jhon**
https://monstrouspeace.com/robintracker/

### CODE REVIEW

**Comment by dashed** [comment]
@jhon not happening. :-1:

**Comment by Bottswana** [comment]
Too many people running robin-grow, dont need that much statistic information when the high percentage of those users will be in the big chat rooms. Only going to cause load issues for all involved.

**Comment by jhon** [comment]
That's easily resolved by rate limiting the tracking beacons. Can easily tune it down so it isn't submitting every 10s.

**Comment by devoidfury** [comment]
See #118 -- interesting idea, but I'd prefer not to add any outbound connections for privacy reasons.

**Comment by SysadminJeroen** [comment]
Could have it off by default, and give the user an option to enable it.

**Comment by Bottswana** [comment]
I wouldn't mind it with the option to enable (default off) and a reduced polling time.

File: robin.user.js [hunk 1]
```diff
@@ -261,6 +266,31 @@
             $robinVoteWidget.find('.robin--vote-class--continue .robin-chat--vote-label').html('stay<br>(' + formatNumber(counts.CONTINUE) + ')');
             users = list.length;
             $(".usercount").text(formatNumber(users) + " users in chat");
+
+            // Given 10s interval, statisticsCounter%6==60s reporting interval
+            if(settings.reportStats && 0==(statisticsCounter%6))
+            {
+                // Report statistics to the automated leaderboard
+                trackers = [
```
Comment (dashed) [inline robin.user.js:288]: Make this a configurable option.

**Comment by dashed** [comment]
I guess this is fine when it's opt-in.

**Comment by jhon** [comment]
The setting in this pull request is to disable it by default and opt-in

**Comment by dkisselev** [comment]
I disagree about disable by default, we want/need new users that are in the lower tiers to report their data, and they're mush less likely to go and click the box to enable reporting.

Maybe set a really slow interval by default (1 min or even 5), but I would say keep it on.

**Comment by jhon** [comment]
opt-in is definitely the more conservative option. In terms of adding this functionality, I'd much rather get the ball rolling opt-in and then have the opt-out conversation if its interesting.

**Comment by dashed** [comment]
@jhon opt-out first and then create an alert/modal/popout to educate people on why they should opt-in

**Comment by jhon** [comment]
I can definitely do that, but the consensus in this thread so far has been opt-in.

That is: The changelist does not report statistics, but will if you toggle the setting.

**Comment by dashed** [comment]
Alright... but either way, a popup is a good idea.

**Comment by jhon** [comment]
If there were already a popup describing new features in this version of the script, I'd do it. However, I don't want to spam the users. I think you're right that it might be useful to add more details but I'd rather add them to the configuration option (linking to the leaderboard that it reports to, etc).

While it won't catch seasoned veterans, having the option there with a description will catch new people who are going through the settings and setting up filters for the first time. In terms of adding the details that @dkisselev is talking about: this is the target audience.

**Comment by dashed** [comment]
@jhon Alright that's fine.

---

People do use configuration option a lot. I'm in T16 with 8+ channels, and peopel switch betwen them; implying the need to switch channels via config screen.

**Comment by jhon** [comment]
Provided more information to the config setting (Mostly acknowledging that it contributes to the the leader board).

**Comment by vartan** [comment]
I think I could get behind this if it's opt-in, can some people verify this PR works and is opt-in?

**Comment by vartan** [comment]
I tried to run this right now, settings did not save and I never saw it update.

I have a feeling something else broke this, not you, but I'm not sure.

**Comment by vartan** [comment]
Once we fix this bug, i'll merge this in.

**Comment by vartan** [comment]
I think the bug was some corruption on my end, not in the script. Tried in a different browser and suddenly it worked.

**Pull request merged**